staging/src/k8s.io/code-generator/cmd/conversion-gen/generators/conversion_test.go

```diff
@@ -0,0 +1,328 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generators
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func Test_extractTag(t *testing.T) {
+	testCases := map[string]struct {
+		comments []string
+		expect   []string
+	}{
+		"no comments": {
+			comments: []string{},
+			expect:   nil,
+		},
+		"no tag": {
+			comments: []string{
+				"+",
+			},
+			expect: nil,
+		},
+		"wrong tag": {
+			comments: []string{
+				"+k8s:wrong-tag",
+			},
+			expect: nil,
+		},
+		"no value": {
+			comments: []string{
+				"+k8s:conversion-gen=",
+			},
+			expect: []string{
+				"",
+			},
+		},
+		"value==false": {
+			comments: []string{
+				"+k8s:conversion-gen=false",
+			},
+			expect: []string{
+				"false",
+			},
+		},
+		"one comment": {
+			comments: []string{
+				"+k8s:conversion-gen=k8s.io/kubernetes/runtime.Object",
+			},
+			expect: []string{
+				"k8s.io/kubernetes/runtime.Object",
+			},
+		},
+		"two different comments": {
+			comments: []string{
+				"+k8s:conversion-gen=k8s.io/kubernetes/runtime.Object",
+				"+k8s:conversion-gen=k8s.io/kubernetes/runtime.List",
+			},
+			expect: []string{
+				"k8s.io/kubernetes/runtime.Object",
+				"k8s.io/kubernetes/runtime.List",
+			},
+		},
+		"two same comments": {
+			comments: []string{
+				"+k8s:conversion-gen=k8s.io/kubernetes/runtime.Object",
+				"+k8s:conversion-gen=k8s.io/kubernetes/runtime.Object",
+			},
+			expect: []string{
+				"k8s.io/kubernetes/runtime.Object",
+				"k8s.io/kubernetes/runtime.Object",
+			},
+		},
+	}
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			r := extractTag(tc.comments)
+			assert.Equal(t, tc.expect, r)
+		})
+	}
+}
+
+func Test_extractExplicitFromTag(t *testing.T) {
```

Comment:
not sure testing all these functions has so much value. They are just one-liner variants of `types.ExtractCommentTags`. The later is unit tested in gengo. Here, we only project to one tag name.

Looking further down in conversion.go, what @aojea meant with coverage in https://github.com/kubernetes/kubernetes/issues/119289 is about the business logic of the generates. For instance `getManualConversionFunctions` is pretty complex. But also the whole package could be unit tested. The functions here in the PR are rather just one or two-liner helpers which are obvious what they do.